SPECS/lld/lld.spec

```diff
@@ -0,0 +1,72 @@
+Summary:        LLD is a linker from the LLVM project that is a drop-in replacement for system linkers and runs much faster than them
+Name:           lld
+Version:        12.0.1
+Release:        1%{?dist}
+License:        NCSA
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          Development/Tools
+URL:            https://clang.llvm.org
+Source0:        https://github.com/llvm/llvm-project/releases/download/llvmorg-%{version}/lld-%{version}.src.tar.xz
+Source1:        https://github.com/llvm/llvm-project/releases/download/llvmorg-%{version}/libunwind-%{version}.src.tar.xz
+BuildRequires:  build-essential
+BuildRequires:  cmake
+BuildRequires:  llvm-devel
+BuildRequires:  ninja-build
+BuildRequires:  python3
+BuildRequires:  file
+
+%package devel
+Summary:        Libraries and header files for LLD
+Requires:       %{name} = %{version}-%{release}
+
+%package libs
+Summary:        LLD shared libraries
+
+%description
+The LLVM project linker.
+
+%description devel
+This package contains library and header files needed to develop new native
+programs that use the LLD infrastructure.
+
+%description libs
+Shared libraries for LLD.
+
+%prep
+%setup -q -b 1 -n lld-%{version}.src
+
+%build
+mkdir -p build
+cd build
+%cmake ..                                                         \
+       -G Ninja                                                   \
+       -DCMAKE_BUILD_TYPE=Release                                 \
+       -DCMAKE_SKIP_RPATH:BOOL=on                                 \
+       -DCMAKE_C_FLAGS=-I../../libunwind-%{version}.src/include   \
+       -DCMAKE_CXX_FLAGS=-I../../libunwind-%{version}.src/include \
+       -DLLVM_LINK_LLVM_DYLIB:BOOL=on                             \
+       -DLLVM_DYLIB_COMPONENTS="all"                              \
+       -Wno-dev
+%ninja_build
+
+%install
+cd build
+%ninja_install
+
+%files
+%license LICENSE.TXT
+%{_bindir}
+
+%files devel
+%{_includedir}/lld
+%{_libdir}/cmake
+%{_libdir}/*.so
+
+%files libs
```

Comment:
Since the libs subpackage is installable independently of the other subpackages, can we package the license here as well?

Reply:
Could do, we don't do that for the `clang` package today which is where I took the split of `main`/`libs` from (normally I'd have expected the `libs` and `main` to be the same package since I don't see a use for the libs outside of the main binary, possibly also applies to the `devel` package...)

Reply:
That would be great. We don't want to accidentally breach any licenses by not redistributing the text of a license. In general, we try to pack the license in the "root" subpackages (I'm sure there's better terminology for this).

For this spec, all of the subpackages transitively depend on the `libs` subpackage.  No matter which RPM you install from this file, you'll also get the `libs` subpackage. So, we only need to pack the license in the `libs` subpackage.

And it looks like we do need to fix the `clang` spec file to put licenses in the right place. I'll put a bug on myself to fix that, thanks!

Reply:
Done